modules/javafx.web/src/test/java/test/javafx/scene/web/LocalStorageTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2011, 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.web;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.concurrent.FutureTask;
+import java.io.File;
+import javafx.scene.web.WebEngineShim;
+import javafx.concurrent.Worker;
+import javafx.scene.web.WebView;
+import javafx.scene.web.WebEngine;
+
+import org.junit.Test;
+
+public class LocalStorageTest extends TestBase {
+
+    @Test
+    public void testLocalStorage() throws Exception {
+        final WebEngine webEngine = getEngine();
+        webEngine.setJavaScriptEnabled(true);
+        webEngine.setUserDataDirectory(new File("/tmp/java-store"));
```

Comment:
You should not hard-code /tmp/. You can either use a local subdirectory in the build dir (which some other tests do), or else you will need to use something like `Files.createTempDirectory("webstorage")`. If you use the latter, then you will need to worry about how to clean up after the test, so the former seems better.

Reply:
i think /tmp/java-store is ok, as it would not require cleanup

Reply:
No, it's not OK for two reasons:

1. It prevents concurrent execution of tests from two different directories
2. The Java temp directory might be something other than "/tmp" on some systems.

Best is to use a local subdir under the build directory.

Reply:
i am unable to find other test case which uses local sub-directory  in the build dir so let me know i can use this
        String defaultBaseDir = System.getProperty("java.io.tmpdir");
        webEngine.setUserDataDirectory(new File(defaultBaseDir + "localstorage")

Reply:
I have updated your suggestions and used local dir for setUserDataDirectory